source/contracts/reporting/ReportingWindow.sol

```diff
@@ -218,6 +228,20 @@ contract ReportingWindow is DelegationTarget, Typed, Initializable, IReportingWi
         return true;
     }
 
+    function collectReportingFees(address _reporterAddress, uint256 _attoReportingTokens) public returns (bool) {
+        IReportingToken _shadyReportingToken = IReportingToken(msg.sender);
+        require(isContainerForReportingToken(_shadyReportingToken));
```

Comment:
Style: I would be OK with inlining the cast in this case: `require(isContainerForReportingToken(IReportingToken(msg.sender)))`.  It is on the boundary of acceptable to me, but the line is so simple that I'm OK with it.  If you prefer the current two-statement style though I'm fine with leaving it as well.